### PR TITLE
Victor/eco 1508/ordering column added to app offers

### PR DESCRIPTION
**Pull request by victordoc**
- [ ] linked to a JIRA issue
- [ ] description of the change
- [ ] tests for the newly added code
- [ ] if this is a breaking change, contains backward support code
- [ ] if this requires client changes or partner updates, updated dependendants
- [ ] were metrics and alerts added
- [ ] should I change documentation

`alter table application_offers add column ordering int;` - will add the column to the prod db, I'll populate the new column manually
`create unique index app_ordering_uniq_idx on application_offers (app_id, ordering)` - unique index for preventing duplicates and sorting

### CODE REVIEW

**Comment by doodyparizada** [comment]
In getOffers you should use the new ordering column for the order

File: scripts/src/models/applications.ts [hunk 1]
```diff
@@ -99,15 +99,22 @@ export class AppOffer extends BaseEntity {
 				.leftJoinAndSelect("app_offer.app", "app")
 				.where("app_offer.appId = :appId", { appId })
 				.andWhere("offer.type = :type", { type })
-				.orderBy("offer.amount", type === "earn" ? "DESC" : "ASC")
-				.addOrderBy("offer.id", "ASC")
+				.orderBy("offer.ordering", "ASC")
```
Comment (doodyparizada) [inline scripts/src/models/applications.ts]: maybe still leave the other orderBy statements as secondary, so when you deploy to production you can leave `ordering` empty to begin with. In any case add the alter table statement here in the PR so we can review it too

**Review by doodyparizada — CHANGES_REQUESTED** [review]

File: scripts/src/models/applications.ts (hunk 1 above)
Comment (victordoc) [reply scripts/src/models/applications.ts]: What I thought is to alter the table first: add a column, populate it with data and afterwards to deploy this fix
How to add the `alter table statement` code to the PR? By a comment in the code or comment in the PR itself?

Comment (doodyparizada) [reply scripts/src/models/applications.ts]: just comment in the PR

Comment (victordoc) [reply scripts/src/models/applications.ts]: Done
I added a unique index to the column for performance and consistency

**Review by doodyparizada — APPROVED** [review]

File: scripts/src/models/applications.ts [hunk 2]
```diff
@@ -99,15 +99,22 @@ export class AppOffer extends BaseEntity {
...
 		}
 
 		return appOffers;
 	}
 
+	public static async generate(appId: string, offerId: string, cap: Cap, walletAddress: string): Promise<AppOffer> {
+		const lastAppOffer = await AppOffer.findOne({ where: { appId }, order: { ordering: "DESC" } });
+		const orderingBufferStep = 10;
+		const lastAppOfferOrdering = (lastAppOffer && lastAppOffer.ordering) ? Number(lastAppOffer.ordering) : 0;
+		const newAppOfferOrdering = lastAppOfferOrdering + 1 * orderingBufferStep;
```
Comment (liorama) [inline scripts/src/models/applications.ts]: I think "OrderIndex" is a better suffix than Ordering, eg:
`newAppOfferOrderIndex`
(This applies to all relevant var names in this context)

File: scripts/src/models/applications.ts [hunk 3]
```diff
@@ -120,6 +127,9 @@ export class AppOffer extends BaseEntity {
 	@Column({ name: "wallet_address" })
 	public walletAddress!: string;
 
+	@Column({ name: "ordering", type: "int" })
+	public ordering!: number;
```
Comment (liorama) [inline scripts/src/models/applications.ts]: Here also, I'd do `orderIndex`

**Pull request merged**